Filter out non-flight type reservations

samdatkins opened the pull request:
Thanks for the great tool! 

I noticed recently I started seeing an error like this:
```Failed to retrieve reservation for {redacted} with confirmation number XXX06390US0. Reason: Bad Request 400.```

Since the confirmation number was clearly not formatted in the typical manner (six characters long), I figured there must be another type of confirmation number that this tool was not taking into account. Sure enough, when looking at the API response it was getting back a reservation that looked something like this:

```{
            "dates": { "first": "2023-02-24", "second": "2023-02-26" },
            "destinationDescription": "Payless, Mid-size - JFK",
            "confirmationNumber": "XXX06390US0",
            "tripType": "CAR",
            ...
}```

This PR fixes the issue for me. I've been running this code for a week or two and I've not encountered any more errors, and check-ins have continued to succeed.

samdatkins commented:
Note that the `XXX` at the start of the confirmation numbers in the above comment are also redacted info, they're actually numbers for the real confirmation number in case that's relevant information.

jdholtz commented:
Thanks for the contribution! I was unaware this could happen, so I’m glad you were able to identify and fix this issue. 

Could you create a test in `tests/test_webdriver.py` for the new changes (let me know if you want me to do this instead)?

samdatkins commented:
You bet, to be honest I was lazily avoiding setting up my local environment because I'm on an M1 machine. Fixed two tests that I broke with my code change and also added a new one to cover the filtering!

jdholtz commented:
Thanks!
